crates/viewer/re_space_view_spatial/src/visualizers/assets3d.rs

```diff
@@ -32,48 +32,53 @@ impl Default for Asset3DVisualizer {
     }
 }
 
-struct Asset3DComponentData {
+struct Asset3DComponentData<'a> {
     index: (TimeInt, RowId),
+    query_result_hash: Hash64,
 
     blob: ArrowBuffer<u8>,
     media_type: Option<ArrowString>,
+    albedo_factor: Option<&'a AlbedoFactor>,
 }
 
 // NOTE: Do not put profile scopes in these methods. They are called for all entities and all
 // timestamps within a time range -- it's _a lot_.
 impl Asset3DVisualizer {
-    fn process_data(
+    fn process_data<'a>(
         &mut self,
         ctx: &QueryContext<'_>,
         render_ctx: &RenderContext,
         instances: &mut Vec<MeshInstance>,
         ent_context: &SpatialSceneEntityContext<'_>,
-        data: impl Iterator<Item = Asset3DComponentData>,
+        data: impl Iterator<Item = Asset3DComponentData<'a>>,
     ) {
         let entity_path = ctx.target_entity_path;
 
         for data in data {
-            let mesh = Asset3D {
-                blob: data.blob.clone().into(),
-                media_type: data.media_type.clone().map(Into::into),
-            };
-
             let primary_row_id = data.index.1;
             let picking_instance_hash = re_entity_db::InstancePathHash::entity_all(entity_path);
             let outline_mask_ids = ent_context.highlight.index_outline_mask(Instance::ALL);
 
             // TODO(#5974): this is subtly wrong, the key should actually be a hash of everything that got
             // cached, which includes the media type…
             let mesh = ctx.viewer_ctx.cache.entry(|c: &mut MeshCache| {
+                let key = MeshCacheKey {
+                    versioned_instance_path_hash: picking_instance_hash.versioned(primary_row_id),
+                    query_result_hash: data.query_result_hash,
+                    media_type: data.media_type.clone().map(Into::into),
+                };
+
                 c.entry(
                     &entity_path.to_string(),
-                    MeshCacheKey {
-                        versioned_instance_path_hash: picking_instance_hash
-                            .versioned(primary_row_id),
-                        query_result_hash: Hash64::ZERO,
-                        media_type: data.media_type.clone().map(Into::into),
+                    key.clone(),
+                    AnyMesh::Asset {
+                        asset: &Asset3D {
+                            blob: data.blob.clone().into(),
+                            media_type: data.media_type.clone().map(Into::into),
+                            albedo_factor: data.albedo_factor.copied(),
+                        },
+                        texture_key: re_log_types::hash::Hash64::hash(&key).hash64(),
```

Comment:
I don't get why this is called texture key when the key is simply about other properties of the Asset3D archetype that we want to have to be part of the cache key. But practically this has nothing to do with textures - neither the intention nor what you're feeding in (which shares some of the exact same parts of `Asset3D` above).
I think the only thing you actually need to do is to populate the `query_result_hash` in the same way as it is done on `Mesh3D` - this key will change every time the albedo color override changes. Everything else can stay the same and no `texture_key` should be needed.